### ISSUE TITLE
aws_availability_zones started proposing changes on each apply

### ISSUE THREAD

**Issue by vladimirtiukhtin** [terraform-providers/terraform-provider-aws#14405]
### Terraform CLI and Terraform AWS Provider Version

```
Terraform v0.13.0-rc1
+ provider registry.terraform.io/-/aws v2.70.0
+ provider registry.terraform.io/-/random v2.3.0
+ provider registry.terraform.io/hashicorp/aws v2.70.0
+ provider registry.terraform.io/hashicorp/random v2.3.0
```

### Affected Resource(s)

* aws_availability_zones

### Debug Output
```
 <= data "aws_availability_zones" "available"  {
        group_names = [
            "eu-west-2",
        ]
      ~ id          = "2020-07-30 12:47:26.4286656 +0000 UTC" -> "2020-07-30 12:47:47.9707994 +0000 UTC"
```

### Expected Behavior

No changes proposed

**Comment by bflad** [comment]
Hi @vladimirtiukhtin 👋  Thank you for submitting this and sorry you ran into trouble here. Can you please provide the relevant data source configuration?

**Comment by vladimirtiukhtin** [comment]
Thanks for the prompt response @bflad. My configuration is incredibly simple
```
data "aws_availability_zones" "available" {
  state = "available"
}
```

**Comment by bflad** [comment]
This should fix it: https://github.com/terraform-providers/terraform-provider-aws/pull/14412

I'm guessing there is going to be other instances of this unexpected behavior with other data sources in Terraform 0.13 as well, but its going to be a week or two before we can easily gain more visibility into this type of issue.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 3.1.0 of the Terraform AWS Provider, next week. 👍

**Comment by vladimirtiukhtin** [comment]
@bflad I'm afraid the issue still exists in 3.1.0. It looks like the root cause is in ID which became dependant on current data/time
<img width="767" alt="Screen Shot 2020-08-10 at 14 33 10" src="https://user-images.githubusercontent.com/15124334/89788096-82e48800-db16-11ea-9455-eb882f1065af.png">

**Comment by vladimirtiukhtin** [comment]
Should we open a new issue or re-open this one?

**Comment by jurgenweber** [comment]
This still seems to be broken:

```
$ tf version
Terraform v0.13.0
+ provider registry.terraform.io/hashicorp/aws v3.1.0
+ provider registry.terraform.io/hashicorp/helm v1.2.4
+ provider registry.terraform.io/hashicorp/kubernetes v1.12.0
+ provider registry.terraform.io/hashicorp/local v1.4.0
+ provider registry.terraform.io/hashicorp/null v2.1.2
+ provider registry.terraform.io/hashicorp/random v2.3.0
+ provider registry.terraform.io/hashicorp/template v2.1.2
+ provider registry.terraform.io/hashicorp/tfe v0.20.0
```

and:
```
 <= data "aws_availability_zones" "available"  {
        group_names = [
            "ap-southeast-2",
        ]
      ~ id          = "2020-08-14 00:15:16.884539377 +0000 UTC" -> "2020-08-14 00:15:48.674203932 +0000 UTC"
        names       = [
            "ap-southeast-2a",
            "ap-southeast-2b",
            "ap-southeast-2c",
        ]
        state       = "available"
        zone_ids    = [
            "apse2-az3",
            "apse2-az1",
            "apse2-az2",
        ]
    }
```

it also seems to be that many data providers are suffering from this; I also see it from:
```
<= data "aws_kms_secrets" "mtribes"  {
      ~ id        = "2020-08-14 00:15:19.986658155 +0000 UTC" -> "2020-08-14 00:15:48.477072138 +0000 UTC"
        plaintext = {
```
and its dumping my secrets to console as plain text!!!!!

also:

https://github.com/terraform-providers/terraform-provider-aws/issues/14649

**Comment by bflad** [comment]
Hi folks 👋 

There was some unexpected changes with data source handling that occurred in the Terraform CLI version 0.13.0 release that are causing this additional data source output in addition to the unstable `id` attribute in the output. There are now larger tracking issues for both sides of this so to consolidate efforts and discussions:

* https://github.com/hashicorp/terraform/issues/25812
* https://github.com/terraform-providers/terraform-provider-aws/issues/14579

Please upvote and subscribe to those for further updates. 👍  I'm going to lock this issue to encourage any discussion in those.
